coderd/provisionerdserver/acquirer_test.go

```diff
@@ -0,0 +1,519 @@
+package provisionerdserver_test
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/goleak"
+	"golang.org/x/exp/slices"
+
+	"cdr.dev/slog"
+	"cdr.dev/slog/sloggers/slogtest"
+	"github.com/coder/coder/v2/coderd/database"
+	"github.com/coder/coder/v2/coderd/database/dbfake"
+	"github.com/coder/coder/v2/coderd/database/pubsub"
+	"github.com/coder/coder/v2/coderd/provisionerdserver"
+	"github.com/coder/coder/v2/testutil"
+)
+
+func TestMain(m *testing.M) {
+	goleak.VerifyTestMain(m)
+}
+
+// TestAcquirer_Store tests that a database.Store is accepted as a provisionerdserver.AcquirerStore
+func TestAcquirer_Store(t *testing.T) {
+	t.Parallel()
+	db := dbfake.New()
+	ps := pubsub.NewInMemory()
+	ctx, cancel := context.WithTimeout(context.Background(), testutil.WaitShort)
+	defer cancel()
+	logger := slogtest.Make(t, nil).Leveled(slog.LevelDebug)
+	_ = provisionerdserver.NewAcquirer(ctx, logger.Named("acquirer"), db, ps)
+}
+
+func TestAcquirer_Single(t *testing.T) {
+	t.Parallel()
+	fs := newFakeOrderedStore()
+	ps := pubsub.NewInMemory()
+	ctx, cancel := context.WithTimeout(context.Background(), testutil.WaitShort)
+	defer cancel()
+	logger := slogtest.Make(t, nil).Leveled(slog.LevelDebug)
+	uut := provisionerdserver.NewAcquirer(ctx, logger.Named("acquirer"), fs, ps)
+
+	workerID := uuid.New()
+	pt := []database.ProvisionerType{database.ProvisionerTypeEcho}
+	tags := provisionerdserver.Tags{
+		"foo": "bar",
+	}
+	acquiree := newTestAcquiree(t, workerID, pt, tags)
+	jobID := uuid.New()
+	go func() {
+		err := fs.sendCtx(ctx, database.ProvisionerJob{ID: jobID}, nil)
+		assert.NoError(t, err)
+	}()
```

Comment:
Add a waitgroup or done chan to ensure this exits before the test closes.

Reply:
Actually, this is from before I made the fake store use buffered channels.  It's safe and simpler to do this on the main goroutine, so I'll fix.